### PR TITLE
gh-108364: In sqlite3, disable foreign keys before dumping SQL schema

### PR DESCRIPTION
**Pull request by felixxm**
Superseded #108471.

* Issue: gh-108364

### CODE REVIEW

File: Lib/sqlite3/dump.py [hunk 1]
```diff
@@ -26,6 +26,10 @@ def _iterdump(connection):
 
     writeable_schema = False
     cu = connection.cursor()
+    # Disable foreign key constraints, if there is any foreign key violation.
+    violations = cu.execute("PRAGMA foreign_key_check").fetchall()
+    if len(violations) > 0:
```
Comment (serhiy-storchaka) [inline Lib/sqlite3/dump.py]: PEP 8
```suggestion
    if violations:
```

File: Lib/test/test_sqlite3/test_dump.py [hunk 2]
```diff
@@ -41,8 +45,12 @@ def test_table_dump(self):
         [self.cu.execute(s) for s in expected_sqls]
         i = self.cx.iterdump()
         actual_sqls = [s for s in i]
-        expected_sqls = ['BEGIN TRANSACTION;'] + expected_sqls + \
-            ['COMMIT;']
+        expected_sqls = [
+            "PRAGMA foreign_keys=OFF;",
+            "BEGIN TRANSACTION;"
+        ] + expected_sqls + [
```
Comment (serhiy-storchaka) [inline Lib/test/test_sqlite3/test_dump.py]: Minor style suggestion.
```suggestion
            "BEGIN TRANSACTION;",
            *expected_sqls,
```

**Review by erlend-aasland — APPROVED** [review]
Thanks, Mariusz!

Please apply Serhiy's style suggestions; I'll land this tomorrow.

**Comment by felixxm** [comment]
Thanks for reviews :1st_place_medal: Updated.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

For future, please do not use rebase and force push. It helps reviewers to see what was changed since the last review.

**Comment by erlend-aasland** [comment]
Thanks, @felixxm!

**Pull request merged**